src/main/java/org/myrobotlab/service/TimeEncoder.java

```diff
@@ -31,28 +51,39 @@ public TimeEncoder(String n) {
   static public ServiceType getMetaData() {
 
     ServiceType meta = new ServiceType(TimeEncoder.class);
-    meta.addDescription("used as a general template");
+    meta.addDescription("general purpose timing encoder used in place of real hardware");
     meta.setAvailable(true);
     meta.setAvailable(false);
     meta.addCategory("general");
     return meta;
   }
 
   @Override // from EncoderController
-  public void attach(EncoderControl control, Integer pin) throws Exception {
+  public EncoderData publishEncoderData(EncoderData data) {
     // TODO Auto-generated method stub
-    
+    return null;
   }
 
-  @Override // from EncoderController
-  public EncoderData publishEncoderData(EncoderData data) {
+  public void attach(EncoderController controller) throws Exception {
     // TODO Auto-generated method stub
-    return null;
+
   }
 
-  @Override // from EncoderController
-  public void setZeroPoint(EncoderControl encoder) {
+  @Override
+  public void disable() {
     // TODO Auto-generated method stub
-    
+
+  }
+
+  @Override
+  public void enable() {
+    // TODO Auto-generated method stub
+
+  }
+
+  @Override
+  public Boolean isEnabled() {
+    // TODO Auto-generated method stub
+    return null;

```

Comment:
needs implementation should not return null.  actually this method really shouldn't return a Boolean.  In general , return values from our interfaces ,  I feel should be primitives , especially in the case of boolean.

Reply:
oye